client/chat.ts

```diff
@@ -0,0 +1,61 @@
+import { sendChatMessage } from './server.js';
+
+// Represents a chat message
+interface Message {
+    id: string;
+    body: string;
+}
+
+// If we need to pre-populate the chat with previous messages, they get
+// loaded from here.
+const chatData: Message[] = [];
+
+/**
+ * Take in a `Message` object and return an HTML element with the chat
+ * message.
+ */
+const formatMessage = (data: Message): HTMLElement => {
+    const div = document.createElement('div');
+    div.classList.add('message');
+    // TODO: the username should eventually be better than a sliced random ID.
+    div.innerHTML = `
+        <span class="username">${data.id.slice(
+            -5
+        )}</span> <span class="message-body">${data.body}</span>
+    `;
+    return div;
+};
+
+/**
+ * Take in a message's HTML element and append it to the DOM.
+ */
+const printMessage = (message): void => {
+    const messageOutputArea = document.getElementById('messages');
+    messageOutputArea.appendChild(message);
+};
+
+/**
+ * Send new messages
+ */
+const initializeChatListener = (socket): void => {
+    const chatForm: HTMLElement = document.getElementById('chatForm');
+
+    chatForm.addEventListener('submit', (e): void => {
+        e.preventDefault();
+        const chatMessageInput = <HTMLInputElement>(
+            document.getElementById('chatMessageInput')
+        );
+        const chatMessage: string = chatMessageInput.value.trim();
+        chatMessageInput.value = '';
+        console.log('from form:', chatMessage);
+        sendChatMessage(socket, chatMessage);
+    });
+
+    socket.on('chat', (message): void => {
+        console.log('from server:', message);
+        const messageHtml = formatMessage(message);
+        printMessage(messageHtml);
+    });
+};
+
+export { chatData, formatMessage, printMessage, initializeChatListener };
```

Comment:
I'm in the habit of doing `export function foo` rather than putting the exports in the bottom. My rationale for doing it that way is that you can see in the declaration if something is exported.

Is there an advantage to doing exports at the end? 

No strong feelings either way.

Reply:
Doesn't matter to me. I'll update it now.

Reply:
> I could pick some nits about the prettier formatting, but it doesn't seem worth it.

Sorry, I didn't see your top message until after I removed the autoformatting. It's mostly back to the way it was before, except that I left the Typescript quotes as double-quotes for consistency.